### DISCUSSION TITLE
Size volume reconstruction

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1179] [Q&A] [answered]
I'm attempting to reconstruct the volume of a spine phantom. I'm using equipment including the Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB. However, after completing the volume reconstruction, the images of the vertebrae appear too closely spaced together, almost as if the spacing between them is incorrect. Could this issue be related to calibration? What other factors might be influencing the outcome?
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/25fc0c46-89e0-445f-a734-5e6d3539ccab)

**Comment by Sunderlandkyl** [comment]
I think that it is likely related to the calibration, it seems like the ultrasound image spacing is too large.

**Reply by LeidyMoraV** [reply]
The spatial calibration? I'm doing it with fcal 2.0 and 9 wires, and it seems like the error is just 8.49mm
```

    <Transform From="Image" To="Probe"
      Matrix="
        0.0537479	0.0862438	0.0254565	-40.52
        -0.132693	0.0316932	0.0452053	228.346
        0.0414189	-0.0324553	0.111789	14.3739
        0	0	0	1"
       Error="8.44557" Date="032224_095342" />
```

**Reply by Sunderlandkyl** [reply]
8.49mm is a fairly large error. Most image calibrations should normally be less than 1mm.
If you record the image + transform sequence in Slicer I can take a closer look.

**Reply by LeidyMoraV** [reply]
I don't know how to record a sequence. However, I did the calibration again and got a 2mm error, even though the vertebraes keep looking the same.

Scene file: https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/EXqF27RXebtDmcaYPePRyVoBZGTcKIqVA6GaglTta6V_xQ?e=eKjQbg
Config file:https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/ES6B6r_FvF5BinkvUp_zrWoBJNrzME2HvV0MIBpIs3a2MA?e=TcuIob

![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/1662fa3a-2869-4833-b247-7a2cf88425b4)

**Reply by LeidyMoraV** [reply]
I've compared it with only the Ascension 3DGtracker config file to see if my config file (server+calibration) is changing the position markers. But it seems that in both cases the distance is almost the same.
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/f7122533-7b2e-4ef5-9b9f-b6c2f96002e2)

**Reply by Sunderlandkyl** [reply]
The MRML file alone is not enough to debug, the scene would need to be saved as an MRB so that it contains all the image data.
You can find information on how to record a sequence here: https://slicer.readthedocs.io/en/latest/user_guide/modules/sequences.html.

**Reply by LeidyMoraV** [reply]
The data is here: https://pennstateoffice365-my.sharepoint.com/:f:/g/personal/lfm5705_psu_edu/Euc_NmkZvkpBjNjDO8OUV8UBCyPQYKAv1oTKQxlxZnqViA?e=t7P5MT
The sequence is named SequenceBrowser_1

**Reply by Sunderlandkyl** [reply]
Ok, I see the problem now. Your Slicer scene is not taking ImageToProbe into account. You'll need to either change the reference coordinate system of your image to Image_Probe, or add the ImageToProbe transform to the outgoing server and adjust your transform hierarchy so that it goes Image_Image -> ImageToProbe -> ProbeToTracker.

You should probably also do the volume reconstruction in the Reference coordinate system in case the phantom is moved accidentally. You can just add ProbeToReference to the config file and use that instead of ProbeToTracker.

**Reply by LeidyMoraV** [reply]
Do you have any config files example of the changes I need to make?

**Reply by Sunderlandkyl** [reply, marked as answer]
Actually I realized there's an easier way. You can just change the parent transform of Image_Image to ImageToReference and use that instead, or use the Image_Reference volume for reconstruction instead.

Just don't try to record a sequence in the future using Image_Reference since the saved sequence will not contain the position information.

___

In any case here's some examples of other changes that you could make:

Change the EmbeddedTransformToFrame to "Probe" so that the image calibration is applied.
```
  <Image Name="Image" EmbeddedTransformToFrame="Probe" />
```

Add ProbeToReference to this section to get the "ProbeToReference" transform in Slicer.
```
      <TransformNames>
        <Transform Name="ProbeToTracker" />
        <Transform Name="ReferenceToTracker" />
        <Transform Name="StylusToTracker" />
        <Transform Name="ImageToReference" />
        <Transform Name="ProbeToReference" />
      </TransformNames>
```

**Reply by LeidyMoraV** [reply]
It worked, thank you!
